test/TestUtilities/Test.Utility/IconTestSourceDirectory.cs

```diff
@@ -0,0 +1,188 @@
+
+            var fileList = new List<Tuple<string, int>>();
+
+            if (!string.IsNullOrEmpty(fileName))
+            {
+                entriesList.Add(Tuple.Create(fileName, string.Empty));
+            }
+
+            if (iconFileSize > 0)
+            {
+                fileList.Add(Tuple.Create(fileName, iconFileSize));
+            }
+
+            Init(fileList, entriesList);
+        }
+
+        /// <summary>
+        /// Constructor for test cases with multiple files and multiple &lt;file /&gt; entries in the nuspec.
+        /// </summary>
+        /// <param name="iconName">&lt;icon /> entry in the nuspec.</param>
+        /// <param name="files">List of (filePath, fileSize) tuples for the test files to create</param>
+        /// <param name="fileEntries">List of strings written as &lt;file src="{entry}" /&gt; </param>
+        public IconTestSourceDirectory(string iconName, IEnumerable<Tuple<string, int>> files, IEnumerable<string> fileEntries)
+        {
+            IconEntry = iconName;
+
+            Init(files, fileEntries.Select(x => Tuple.Create(x, string.Empty)));
+        }
+
+        /// <summary>
+        /// Constructor for test cases with multiple files and multiple &lt;file /&gt; entries with target attribute.
+        /// </summary>
+        /// <param name="iconName">&lt;icon /> entry in the nuspec.</param>
+        /// <param name="files">List of (filePath, fileSize) tuples for the test files to create</param>
+        /// <param name="fileEntries">List of (src, tgt) tuples written as &lt;file src="{src}" target="{tgt}" /&gt; </param>
+        public IconTestSourceDirectory(string iconName, IEnumerable<Tuple<string, int>> files, IEnumerable<Tuple<string, string>> fileEntries)
+        {
+            IconEntry = iconName;
+
+            Init(files, fileEntries);
+        }
+
+        private void Init(IEnumerable<Tuple<string, int>> files, IEnumerable<Tuple<string, string>> fileEntries)
+        {
+            TestDirectory = TestDirectory.Create();
+
+            CreateFiles(files);
+            CreateNuspec(fileEntries);
+        }
+
+        private void CreateFiles(IEnumerable<Tuple<string, int>> files)
+        {
+            foreach (var f in files)
+            {
+                var filepath = Path.Combine(BaseDir, f.Item1);
+                var dir = Path.GetDirectoryName(filepath);
+
+                Directory.CreateDirectory(dir);
+                using (var fileStream = File.OpenWrite(Path.Combine(BaseDir, f.Item1)))
+                {
+                    fileStream.SetLength(f.Item2);
+                }
+            }
+        }
+
+        private void CreateNuspec(IEnumerable<Tuple<string, string>> fileEntries)
+        {
+            StringBuilder sb = new StringBuilder();
+            var nl = Environment.NewLine;
+
+            sb.Append("<?xml version=\"1.0\" encoding=\"utf-8\"?>").Append(nl);
+            sb.Append("<package>").Append(nl);
```

Comment:
Why not just use `.AppendLine()`?

Reply:
Because `.Append(nl)` is shorter than `.AppendLine()` but, that would save me from managing the newline by myself

Reply:
`sb.AppendLine("<package>")` is shorter still.

Reply:
I will change to `AppendLine()`